pkg/server/secrets-encrypt.go

```diff
@@ -0,0 +1,489 @@
+package server
+
+import (
+	"bytes"
+	"crypto/rand"
+	"crypto/sha256"
+	"encoding/base64"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"text/tabwriter"
+	"time"
+
+	"github.com/rancher/k3s/pkg/daemons/config"
+	"github.com/rancher/wrangler/pkg/generated/controllers/core"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	apiserverconfigv1 "k8s.io/apiserver/pkg/apis/config/v1"
+)
+
+const (
+	EncryptionStart     string = "start"
+	EncryptionPrepare   string = "prepare"
+	EncryptionRotate    string = "rotate"
+	EncryptionReencrypt string = "reencrypt"
+)
+
+const aescbcKeySize = 32
+
+type EncryptionState struct {
+	Stage      string `json:"stage"`
+	CurrentKey apiserverconfigv1.Key
+}
+
+type EncryptionRequest struct {
+	Stage  string `json:"stage,omitempty"`
+	Toggle bool   `json:"toggle,omitempty"`
+	Force  bool   `json:"force"`
+}
+
+func getEncryptionRequest(req *http.Request) (EncryptionRequest, error) {
+	b, err := ioutil.ReadAll(req.Body)
+	if err != nil {
+		return EncryptionRequest{}, err
+	}
+	result := EncryptionRequest{}
+	err = json.Unmarshal(b, &result)
+	return result, err
+}
+
+func encryptionStatusHandler(server *config.Control) http.Handler {
+	return http.HandlerFunc(func(resp http.ResponseWriter, req *http.Request) {
+		if req.TLS == nil {
+			resp.WriteHeader(http.StatusNotFound)
+			return
+		}
+		status, err := encryptionStatus(server)
+		if err != nil {
+			resp.WriteHeader(http.StatusBadRequest)
+			resp.Write([]byte(err.Error()))
+			return
+		}
+		resp.Write([]byte(status))
+	})
+}
+
+func encryptionStatus(server *config.Control) (string, error) {
+	providers, err := getEncryptionProviders(server)
+	if os.IsNotExist(err) {
+		return "Encryption Status: Disabled, no configuration file found", nil
+	} else if err != nil {
+		return "", err
+	}
+	var statusOutput string
+	if providers[1].Identity != nil && providers[0].AESCBC != nil {
+		statusOutput += "Encryption Status: Enabled\n"
```

Comment:
Return a filled `SecretsEncryptionStatus` here instead of just human-readable strings

Reply:
Converted it to return a status struct and do the string formatting on the client side